kube-runtime/src/wait.rs

```diff
@@ -237,6 +242,69 @@ pub mod conditions {
         }
     }
 
+    /// An await condition for `Deployment` that returns `true` once the latest deployment has completed
+    ///
+    /// This looks for the condition that Kubernetes sets for completed deployments:
+    /// <https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#complete-deployment>
+    #[must_use]
+    pub fn is_deployment_completed() -> impl Condition<Deployment> {
+        |obj: Option<&Deployment>| {
+            if let Some(depl) = &obj {
+                if let Some(s) = &depl.status {
+                    if let Some(conds) = &s.conditions {
+                        if let Some(dcond) = conds.iter().find(|c| {
+                            c.type_ == "Progressing" && c.reason == Some("NewReplicaSetAvailable".to_string())
+                        }) {
+                            return dcond.status == "True";
+                        }
+                    }
+                }
+            }
```

Comment:
We have technically relied on integration strategy for testing some other conditions, but i think we can have an easier unit test strategy with dumping some inline yaml into a unit test ala https://github.com/kube-rs/kube/blob/437821285df1d91d0ed5956b8d088056f026e922/kube-core/src/crd.rs#L143
deserialize it into a `Deployment` and verify that the function returns true for it.

Reply:
Roger that, I'll whip up some unit tests with that.

Reply:
tests added!

Reply:
looks great, thank you!